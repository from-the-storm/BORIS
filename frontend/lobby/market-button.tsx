import * as React from 'react';
import { bind } from 'bind-decorator';

import { MarketStatus } from '../../common/api';

import * as openImg from './images/market/open.png';
import * as tapedImg from './images/market/move.png';

interface MarketImage { url: string; id: string; }
const open: MarketImage = { url: openImg, id: 'open' };
const taped: MarketImage = { url: tapedImg, id: 'move' };

interface Props {
    status: MarketStatus;
    onClick: () => void;
}

export class MarketButton extends React.PureComponent<Props> {
    render() {
        const image = this.image;
        return (
            <div id="market-button-container">
                <button className="market-button" id={image.id} onClick={this.handleMarketClicked}>
                    <img src={image.url} alt="Market Button" />
                </button>
            </div>
        )
    }
    @bind private handleMarketClicked() {
        this.props.onClick();
    }
    private get image(): MarketImage {
        switch (this.props.status) {
            case MarketStatus.AlreadyBought: return taped;
            case MarketStatus.Open: return open;
            case MarketStatus.Taped: return taped;
            default: return {url: "", id: ""};
        }
    }
}
