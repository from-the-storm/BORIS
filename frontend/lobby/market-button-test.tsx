import 'jest';
import * as React from 'react';
import {Â MarketButton } from './market-button';
import * as renderer from 'react-test-renderer';
import * as prettyFormat from 'pretty-format';
const { ReactTestComponent } = prettyFormat.plugins;

const jsonToString = (rendered: any) => prettyFormat(rendered, { plugins: [ReactTestComponent],});

describe("<MarketButton> Component", () => {
    for (let completedScenarios of [1, 2]) {
        describe(`When ${completedScenarios} scenario(s) have been completed`, () => {
            test(`Shows 'Market is Open' when user is The Burdened`, () => {
                const component = renderer.create(
                    <MarketButton completedScenarios={completedScenarios} isBurdened={true} />
                );
                const componentRendered = component.toJSON();
                expect(jsonToString(componentRendered)).toContain('id="open"');
                expect(jsonToString(componentRendered)).not.toContain('id="actualized"');
                expect(componentRendered).toMatchSnapshot();
            });
            test(`Shows 'Move Along' market image when user is not The Burdened`, () => {
                const component = renderer.create(
                    <MarketButton completedScenarios={completedScenarios} isBurdened={false} />
                );
                const componentRendered = component.toJSON();
                expect(jsonToString(componentRendered)).toContain('id="move"');
                expect(componentRendered).toMatchSnapshot();
            });
        });
    }
    for (let completedScenarios of [3, 4]) {
        describe(`When ${completedScenarios} scenario(s) have been completed`, () => {
            test(`Shows "actualized" image when user is The Burdened`, () => {
                const component = renderer.create(
                    <MarketButton completedScenarios={completedScenarios} isBurdened={true} />
                );
                const componentRendered = component.toJSON();
                expect(jsonToString(componentRendered)).toContain('id="actualized"');
                expect(componentRendered).toMatchSnapshot();
            });
            test(`Shows 'Market is Closed' market image when user is not The Burdened`, () => {
                const component = renderer.create(
                    <MarketButton completedScenarios={completedScenarios} isBurdened={false} />
                );
                const componentRendered = component.toJSON();
                expect(jsonToString(componentRendered)).toContain('id="closed"');
                expect(componentRendered).toMatchSnapshot();
            });
        });
    }
});
