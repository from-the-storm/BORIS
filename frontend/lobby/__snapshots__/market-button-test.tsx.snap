// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarketButton> Component When 1 scenario(s) have been completed Shows 'Market is Open' when user is The Burdened 1`] = `
<div
  id="market-button-container"
>
  <button
    className="market"
    id="open"
  >
    <img
      alt="Market Button"
      src="frontend/lobby/images/market/open-HASH.png"
    />
  </button>
</div>
`;

exports[`<MarketButton> Component When 1 scenario(s) have been completed Shows 'Move Along' market image when user is not The Burdened 1`] = `
<div
  id="market-button-container"
>
  <button
    className="market"
    id="move"
  >
    <img
      alt="Market Button"
      src="frontend/lobby/images/market/move-HASH.png"
    />
  </button>
</div>
`;

exports[`<MarketButton> Component When 2 scenario(s) have been completed Shows 'Market is Open' when user is The Burdened 1`] = `
<div
  id="market-button-container"
>
  <button
    className="market"
    id="open"
  >
    <img
      alt="Market Button"
      src="frontend/lobby/images/market/open-HASH.png"
    />
  </button>
</div>
`;

exports[`<MarketButton> Component When 2 scenario(s) have been completed Shows 'Move Along' market image when user is not The Burdened 1`] = `
<div
  id="market-button-container"
>
  <button
    className="market"
    id="move"
  >
    <img
      alt="Market Button"
      src="frontend/lobby/images/market/move-HASH.png"
    />
  </button>
</div>
`;

exports[`<MarketButton> Component When 3 scenario(s) have been completed Shows "actualized" image when user is The Burdened 1`] = `
<div
  id="market-button-container"
>
  <button
    className="market"
    id="actualized"
  >
    <img
      alt="Market Button"
      src="frontend/lobby/images/market/actualized-HASH.png"
    />
  </button>
</div>
`;

exports[`<MarketButton> Component When 3 scenario(s) have been completed Shows 'Market is Closed' market image when user is not The Burdened 1`] = `
<div
  id="market-button-container"
>
  <button
    className="market"
    id="closed"
  >
    <img
      alt="Market Button"
      src="frontend/lobby/images/market/closed-HASH.png"
    />
  </button>
</div>
`;

exports[`<MarketButton> Component When 4 scenario(s) have been completed Shows "actualized" image when user is The Burdened 1`] = `
<div
  id="market-button-container"
>
  <button
    className="market"
    id="actualized"
  >
    <img
      alt="Market Button"
      src="frontend/lobby/images/market/actualized-HASH.png"
    />
  </button>
</div>
`;

exports[`<MarketButton> Component When 4 scenario(s) have been completed Shows 'Market is Closed' market image when user is not The Burdened 1`] = `
<div
  id="market-button-container"
>
  <button
    className="market"
    id="closed"
  >
    <img
      alt="Market Button"
      src="frontend/lobby/images/market/closed-HASH.png"
    />
  </button>
</div>
`;
